src/adapter/src/catalog/storage/stash.rs

```diff
@@ -64,43 +81,22 @@ const DEFAULT_REPLICA_ID: u64 = 1;
 const DEFAULT_REPLICA_NAME: &str = "r1";
 
 const MATERIALIZE_DATABASE_ID_VAL: u64 = 1;
-const MATERIALIZE_DATABASE_ID: DatabaseId = DatabaseId::System(MATERIALIZE_DATABASE_ID_VAL);
```

Comment:
This wasn't used anywhere yet. To make sure the new initialization code did the same thing as the old code, what I ended up doing was writing a snapshot test using [`insta`](https://insta.rs/), snapshotting the debug repr of a `catalog::Transaction`, and then running this branch against that snapshot. 

I didn't add that test to this PR because it was >1000 lines since it includes the snapshot, but I plan on putting it up after this merges if we think there's value.